File: spec/src/utils/orientationUtils.ts
import {Orientation} from '../types/sector';

/** Affichage court avec flèche directionnelle — pour l'UI (badge, liste). */
export const ORIENTATION_LABEL: Record<Orientation, string> = {
  N:  '↑ N',
  NE: '↗ NE',
  E:  '→ E',
  SE: '↘ SE',
  S:  '↓ S',
  SW: '↙ SW',
  W:  '← W',
  NW: '↖ NW',
};

/** Nom complet en français — pour les messages de notification. */
export const ORIENTATION_FR: Record<Orientation, string> = {
  N:  'Nord',
  NE: 'Nord-Est',
  E:  'Est',
  SE: 'Sud-Est',
  S:  'Sud',
  SW: 'Sud-Ouest',
  W:  'Ouest',
  NW: 'Nord-Ouest',
};

File: app/src/main/java/com/almostblue/domain/OrientationUtils.kt
package com.almostblue.domain

/** Port de spec/src/utils/orientationUtils.ts. */

/** Affichage court avec flèche directionnelle — pour l'UI (badge, liste). */
val Orientation.label: String
    get() = when (this) {
        Orientation.N -> "↑ N"
        Orientation.NE -> "↗ NE"
        Orientation.E -> "→ E"
        Orientation.SE -> "↘ SE"
        Orientation.S -> "↓ S"
        Orientation.SW -> "↙ SW"
        Orientation.W -> "← W"
        Orientation.NW -> "↖ NW"
    }

/** Nom complet en français — pour les messages de notification. */
val Orientation.frenchName: String
    get() = when (this) {
        Orientation.N -> "Nord"
        Orientation.NE -> "Nord-Est"
        Orientation.E -> "Est"
        Orientation.SE -> "Sud-Est"
        Orientation.S -> "Sud"
        Orientation.SW -> "Sud-Ouest"
        Orientation.W -> "Ouest"
        Orientation.NW -> "Nord-Ouest"
    }
